www/js/main.js

```diff
@@ -69,19 +73,63 @@ define([
 			try{
 				_model.loadModel(solutionGraph);
 			} catch(err){
-				throw Error(err);
+				if (query.m == "AUTHOR") {
+					//var errorMessage = new messageBox("errorMessageBox", "error", error.message);
+					//errorMessage.show();
+					throw Error(err);
+				} else {
+					//var errorMessage = new messageBox("errorMessageBox", "error", "This problem could not be loaded. Please contact the problem's author.");
+					//errorMessage.show();
+					throw Error("Model could not be loaded.");
+				}
+			}
+			// This version of code addresses loading errors in cases where problem is empty, incomplete or has no root node in coached mode
+			if (query.m !== "AUTHOR") {
```

Comment:
I think using this here is still fine, as main.js is something like our direct connection to that request response. So using the query parameter is fine here. But draw model shouldnt be directly effected by the query parameters. So let's set it in the model.js and use that from now going forward? And with time we can change it at other places as well. I think I have made this same mistake a lot many times.